frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.unit.spec.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { getIcon } from "__support__/ui";
+import { createMockState } from "metabase-types/store/mocks";
+import { createEntitiesState } from "__support__/store";
+import {
+  createSampleDatabase,
+  ORDERS_ID,
+} from "metabase-types/api/mocks/presets";
+import { getMetadata } from "metabase/selectors/metadata";
+import ExpressionWidget, { ExpressionWidgetProps } from "./ExpressionWidget";
+
+describe("ExpressionWidget", () => {
+  it("should render proper controls", () => {
+    setup();
+
+    expect(screen.getByText("Expression")).toBeInTheDocument();
+    expect(screen.getByText("Name")).toBeInTheDocument();
+
+    expect(screen.getByText("Cancel")).toBeInTheDocument();
+    expect(screen.getByText("Done")).toBeInTheDocument();
+  });
+
+  it("should render help icon with tooltip", () => {
+    setup();
+
+    const icon = getIcon("info");
+    expect(icon).toBeInTheDocument();
+
+    userEvent.hover(icon);
+
+    expect(
+      screen.getByText(
+        "You can reference columns here in functions or equations, like: floor([Price] - [Discount]).",
+      ),
+    ).toBeInTheDocument();
+  });
+});
+
+const createMockQueryForExpressions = () => {
+  const state = createMockState({
+    entities: createEntitiesState({
+      databases: [createSampleDatabase()],
+    }),
+  });
+
+  const metadata = getMetadata(state);
+  const query = metadata.table(ORDERS_ID)?.query();
+
+  return query;
+};
+
+const onChangeExpressionMock = jest.fn();
```

Comment:
I think we usually return a mock function from `setup`, like:
```
const setup() {
  const onChangeExpression = jest.fn();

  renderWithProviders(...);

  return { onChangeExpression }; 
});
```

Reply:
I'm also fine if it was provided from the place where `setup` is called, but a global mock looks like something we don't usually do